src/dal/interface/index.ts

```diff
@@ -13,20 +13,39 @@ export class DbClient {
     private db?: Db;
 
     constructor(args: IDbClient) {
-        this.uri = `${args.url}:${args.port}`;
+        this.uri = `${args.url}:${args.port}/${args.dbName}`;
         this.dbName = args.dbName;
     }
 
     // tslint:disable:no-floating-promises
     public async connect(): Promise<void> {
         let client: MongoClient;
         try {
-            client = await MongoClient.connect(this.uri);
+            client = await MongoClient.connect(this.uri, { auth: { user: 'root', password: 'example' } });
             this.db = client.db(this.dbName);
-        } catch(err) {
+        } catch (err) {
             throw err;
         }
-        client.close();
+    }
+
+    public async insert(json: object[]) {
+        let result: InsertWriteOpResult;
+        // tslint:disable:no-any
+        let collections: Collection<any>[];
+        if (this.db) {
```

Comment:
Shouldn't we have a condition here to catch the case when `this.db` doesn't exist?

Reply:
I'm throwing an error now from this function. We'll probably want to handle it later up the pipeline in the request handler.